### ISSUE TITLE
Allow to exclude fields in search

### ISSUE THREAD

**Issue by eric-saintetienne** [JonnyWong16/plexpy#2408]
### Describe the Bug

Hi, I've got a relatively large history (in excess of a hundred pages) and would like to search the history for whether/when a user has seen the TV show named `Rome`.

The problem is that tautulli returns the whole history (the 100 pages) because every row has fields that contain the words `rome`:
* the `Player` field is `Chromecast`
* `Platform` is also `Chromecast`

I've tried these with very limited success:
* Not include these columns in the set of columns that are displayed
* Enclose `rome` with double quotes

Note that somewhat related, this issue is different from #1645 (ability to exclude users in search)

### Technical details about the installation

* Tautulli Version: `2.14.5`
* Git Branch: [`master`](https://github.com/Tautulli/Tautulli/tree/master)
* Git Commit Hash: [`v2.14.5`](https://github.com/Tautulli/Tautulli/commit/v2.14.5)
* Platform and Version: `[Docker] Linux 6.10.9-200.fc40.x86_64 (#1 SMP PREEMPT_DYNAMIC Sun Sep 8 17:23:55 UTC 2024 - Alpine Linux 3.20.3)`
* Python Version: `3.12.6 (main, Sep 11 2024, 08:48:26) [GCC 13.2.1 20240309]`
* Browser and Version: `Firefox 130`

**Comment by eric-saintetienne** [comment]
I could manage using SQL but that's not something I want to use on a regular basis because it's a pain and the schema may change. Other users may not be tech savvy enough to open sqlite and come up with a SQL query like that one:

```
SELECT full_title, date(started, 'unixepoch'), date(stopped, 'unixepoch') FROM session_history
       JOIN session_history_metadata ON reference_id = session_history_metadata.id
       WHERE grandparent_title = 'Rome';
```

A proper fix would be either to select the fields the search operates on in the UI (either globally in the settings page or individually on the history/search page) or biff-up the search, for example using regex with the option to enable case sensitivity.

**Comment by JonnyWong16** [comment]
At least for this specific scenario

> would like to search the history for whether/when a user has seen the TV show named `Rome`

You can just go to the media info page for the TV show which will only show history for that show.

**Comment by eric-saintetienne** [comment]
Thatnks for the suggestion but that'd be a workaround, not a fix.

Unfortunately it doesn't seem to work: I searched for `Rome` in the `All Libraries` tab and nothing came up. I think it's because the show has been deleted after it's been watched so it's not in the current Plex library any longer.

**Comment by JonnyWong16** [comment]
You just click on the show in the main history table, then click the breadcrumb at the top to go up to the show level. Might need to add `&source=history` to the URL.

**Comment by eric-saintetienne** [comment]
In the History table, if I click on `TV Shows` only TV shows are displayed, but there are still 75 pages of them to go through. I'm afraid that's not practical.

**Comment by JonnyWong16** [comment]
I'm referring to the breadcrumbs.

![image](https://github.com/user-attachments/assets/a65e0e8e-7bbd-40e4-afaa-dce5ee487465)

**Comment by eric-saintetienne** [comment]
It looks great on your screenshot, but actually it still doesn't work: I can click on a show that is on the first page, say `The Sopranos`:

<img src="https://github.com/user-attachments/assets/7e1d3a81-53c3-4573-896b-3a55e0429631" width="75%">

But to get to other TV shows, I need to click on the top level item `TV Shows` (`Séries` in my screenshot) and then there's no way to get to a show that's been watched years ago:

<img src="https://github.com/user-attachments/assets/572d1d2c-1b00-40fc-83eb-a0b0f5c7b154" width="75%">

In this screenshot, clickling the `History` tab shows a view that has the same problem of `Rome` matching unrelated fields like `Player`=`Chromecast`.

Please do try to get to an old show from that page, especially one that isn't in your library any longer. You won't be able to because only the recent shows are directly accessible from this page.

**Comment by JonnyWong16** [comment]
Yeah I understand, you would have to find at least one entry for the show `Rome` in the history table to click on.

**Comment by eric-saintetienne** [comment]
Indeed and thanks for looking into it.

It would be straightforward if the show was still present in Plex's library, by clicking `Media Info`. The list in there isn't long to browse, and the search functionality would work better as there's no field like `Player` or `Platform`, although there are other fields though but they all are empty in my case.

How about adding an option in `Media Info` to display shows that are not present in Plex's library any longer?

**Comment by JonnyWong16** [comment]
Merging the live Plex library media info with the Tautulli history media info is probably more difficult than changing the table search.
